src/main/java/org/circle8/dao/PuntoReciclajeDao.java

```diff
@@ -37,53 +56,150 @@ public class PuntoReciclajeDao extends Dao{
+		    DELETE FROM public."PuntoReciclaje" AS pr
+		    	WHERE pr."ID" = ?
+		    	AND pr."CiudadanoId" = ?
+		""";
+
+	private static final String DELETE_RELACION = """
+		    DELETE FROM public."PuntoReciclaje_TipoResiduo" AS r
+		    	WHERE r."PuntoReciclajeId" = ?
+		""";
+
+	private static final String UPDATE = """
+		   UPDATE public."PuntoReciclaje"
+		   	SET 
+		""";
+
+	private static final String WHERE_ID_AND_CIUDADANO = """
+		   WHERE "ID" = ?
+		    	AND "CiudadanoId" = ?
+		""";
+
 	@Inject
 	PuntoReciclajeDao(DataSource ds) {
 		super(ds);
 	}
 
+
+	public PuntoReciclaje save(Transaction t, PuntoReciclaje punto) throws PersistenceException {
+
+		try ( var insert = t.prepareStatement(INSERT_SQL, Statement.RETURN_GENERATED_KEYS) ) {
+			insert.setLong(1, punto.recicladorId);
+			insert.setDouble(2, punto.latitud);
+			insert.setDouble(3, punto.longitud);
+			insert.setString(4, Dia.getDias(punto.dias));
+			insert.setString(5, punto.titulo);
+
+			int insertions = insert.executeUpdate();
+			if ( insertions == 0 )
+				throw new SQLException("Creating the PuntoReciclaje failed, no affected rows");
+
+			try ( var rs = insert.getGeneratedKeys() ) {
+				if (rs.next())
+					punto.id = rs.getLong(1);
+				else
+					throw new SQLException("Creating the PuntoReciclaje failed, no ID obtained");
+			}
+		} catch ( SQLException e ) {
+				throw new PersistenceException("error inserting PuntoReciclaje", e);
+		}
+
+		return punto;
+	}
+
 	/**
 	 * Obtiene el listado de puntos de reciclaje
 	 */
-	public List<PuntoReciclaje> list(PuntoReciclajeFilter filter) throws PersistenceException{
-		try ( var t = open(true); var select = createSelectForList(t, filter) ) {
-			try ( var rs = select.executeQuery() ) {
+	public List<PuntoReciclaje> list(PuntoReciclajeFilter filter) throws PersistenceException {
+		try (var t = open(true); var select = createSelectForList(t, filter)) {
+			try (var rs = select.executeQuery()) {
 				return getList(rs, filter);
 			}
-		} catch ( SQLException e ) {
+		} catch (SQLException e) {
+			throw new PersistenceException("error getting punto reciclaje", e);
+		}
+	}
+
+	/**
+	 * Obtiene un punto de reciclaje por medio de su id
+	 */
+	public Optional<PuntoReciclaje> get(Long id, Long recicladorId) throws PersistenceException {
+		try (var t = open(true); var select = createSelectForGet(t, id, recicladorId);
+			 var rs = select.executeQuery()) {
+			return Optional.of(getPunto(rs));
+		} catch (SQLException e) {
 			throw new PersistenceException("error getting punto reciclaje", e);
 		}
 	}
 
+	/**
+	 * Actualiza un punto de reciclaje
+	 * @param id
+	 * @param recicladorId
+	 * @return
+	 * @throws PersistenceException
+	 */
+	public boolean put(Transaction t, Long id, Long recicladorId, PuntoReciclajeRequest req) throws SQLException, PersistenceException {
+
+		try (var update = createSelectForPut(t, id, recicladorId, req)) {
+
+			if (update.executeUpdate() <= 0)
+				return false;
```

Comment:
esto no es un return false, esto es un error por una razon:
- No encontro lo que iba a borrar -> no es un false, es un not found